### DISCUSSION TITLE
ERC20 Deposit/Withdraw - viem/op-stack

### DISCUSSION THREAD

**Discussion by danielsimao** [wevm/viem#2250] [Idea / Feature Request]
By looking at the docs and some of the source code, I could not find anything related to depositing or withdrawing ERC20 on viem/op-stack. I know there is this functionality in op-viem from base-org. Could someone clarify me if I'm missing something or if this feature is yet to come?

Thank you so much in advance.

**Comment by danielsimao** [comment]
With this feature would be good to also have a fee estimate for bridging erc20.

**Comment by gndelia** [comment]
If you need an example, you can use the following implementation

Using the Portal contract did not work, at least with the old version of OP. So I had to use a custom implementation for depositing and initiating withdrawals for ERC20, using the L1StandardBridge, and the L2Bridge. In case anyone needs it, [here is the deposit](https://github.com/hemilabs/ui-monorepo/blob/05e51c5965e70f72993332945c357ba1746ec5f2/packages/hemi-tunnel-actions/src/depositErc20.ts) and the [withdrawal](https://github.com/hemilabs/ui-monorepo/blob/main/packages/hemi-tunnel-actions/src/initiateWithdraw.ts).

We may eventually move this into a publishable package
